### ISSUE TITLE
Problems with python dependency resolution at run-time in module.nix

### ISSUE THREAD

**Issue by cleeyv** [ngi-nix/liberaforms#1]
A lot of what I did today was figuring out how to make run-time dependencies available to gunicorn/flask when it is started by a systemd service. For build-time, all the dependency resolution is handled by mach-nix in [flake.nix](https://github.com/ngi-nix/liberaforms/blob/flake/flake.nix) , but this doesn't help with the run-time dependencies in  [module.nix](https://github.com/ngi-nix/liberaforms/blob/flake/nix/module.nix). I was able to make it work for a few initial dependencies by pulling the required python modules from nixpkgs into the service environment (penv -> PYTHONPATH), until I got to one that is not packaged in nixpkgs (flask_session). 

I have considered various ways forward but none of them seem very good:

- I started trying to implement mach-nix a second time inside of module.nix but couldn't get it to work. 
- I have a hunch that if I just put the entire module.nix directly into the flake.nix instead of using an import, then variables defined by mach-nix will come in scope for the module and there will more solutions possible. However, this would be a big refactor and seems like there should be a better solution than such a major workaround. 
- I could try manually packaging flask_session in module.nix but I'm pretty sure that I'm going to run into more missing run-time dependencies after that one is fixed, and some of them will also [not be in nixpkgs](https://github.com/ngi-nix/ngi/issues/233#issuecomment-892170818). This is the slow/tedious way I've been doing this so far. It will speed up once I get the hang of it, but it seems like there should be a better way.

Before continuing I wanted to get feedback on this problem and see if anyone sees any other possible solutions that I'm missing.

**Comment by tomberek** [comment]
an example approach: https://github.com/ngi-nix/liberaforms/pull/2
The services starts, but that's as far as I checked it.

**Comment by cleeyv** [comment]
I merged in #2 and it resolved the problem I was having, and I talked to @tomberek about it afterwards to understand what he had done and why it worked. The key piece I was missing before was how the overlay works in this flake.nix (which I had adapted from https://github.com/ngi-nix/vulnerablecode/blob/main/etc/nix/flake.nix but not fully understood):
```
      # A Nixpkgs overlay.
      overlay = final: prev:
        with final.pkgs; {
```
This is the beginning of the overlay section that contains the creation of liberaforms-env (formerly pythonEnv - the renaming is good because it clarifies what this is) by mach-nix. This make it available in the overlay version of pkgs. This is then made available in the module.nix through a key overlay line in the section of the flake.nix where the `nixConfigurations.container.modules` is defined (within which the module.nix is imported):
```
nixpkgs.overlays = [ self.overlay ];
```
So the `pkgs` in module.nix is the one from this nixpkgs overlay, which contains the liberaforms-env from mach-nix, which is how he was successfully able to invoke pkgs.liberaforms-env in the module.nix. Using this environment to run gunicorn then gives it access to all the paths of the required run-time dependencies. As I had hoped, all of the one-by-one dependency definitions I had been manually doing as a inadequate reconstructions of the full liberaforms python environment are no longer necessary!
